public/app/features/explore/LogsVolumePanelList.tsx

```diff
@@ -55,10 +58,24 @@ export const LogsVolumePanelList = ({
     return !isLogsVolumeLimited(data) && zoomRatio && zoomRatio < 1;
   });
 
+  const timeoutError =
+    logsVolumeData?.error?.message?.includes('timeout') || logsVolumeData?.error?.message?.includes('exceeded');
```

Comment:
i understand that right now there is no better way, but this check binds the solution very directly to datasource=loki, and to grafana-cloud (on-prem installations might use a different error message)

Reply:
That is correct. The other error format that I've seen is `{"status":"error","errorType":"timeout","error":"context deadline exceeded"}` from Cloud instances.

In any case, we've been asked to improve this, so even if not ideal (Progress over perfection), it would be nice to at least take an initial step towards an improved handling of this error.

I will use your feedback to work on an alternative version of this that better aligns with our expectations. Thanks!